pkg/segment/query/segquery.go

```diff
@@ -857,18 +856,8 @@ func applyAggOpOnSegments(sortedQSRSlice []*QuerySegmentRequest, allSegFileResul
 		// If agg has evaluation functions, we should recompute raw data instead of using the previously stored statistical data in the segment
 
 		var sstMap map[string]*structs.SegStats
-		if canUseSSTForStats(searchType, isSegmentFullyEnclosed, segReq.aggs) {
-			sstMap, err = segread.ReadSegStats(segReq.segKey, segReq.qid)
-			if err != nil {
-				log.Errorf("qid=%d,  applyAggOpOnSegments : ReadSegStats: Failed to get segment level stats for segKey %+v! Error: %v", qid, segReq.segKey, err)
-				allSegFileResults.AddError(err)
-				continue
-			}
-			sstMap["*"] = &structs.SegStats{
-				Count: uint64(segReq.TotalRecords),
-			}
-			allSegFileResults.AddResultCount(uint64(segReq.TotalRecords))
-		} else {
+
+		computeSegStatsFunc := func() {
```

Comment:
nit: why not just have this as a standalone func instead of a closure ? makes reading code easier/modular as well func length smaller (the parent's)

Reply:
addressed in the latest commit.